### ISSUE TITLE
Issue with has_many Relationship Not Loading Correctly in Show View

### ISSUE THREAD

**Issue by abdellahi-brahim** [avo-hq/avo#1796]
### Steps to Reproduce
- Create 2 models, with a has_many or has_and_belongs_to_many relationship
- In the index represent the relationship with tags and in the show with the corresponding association field

### Expected behavior & Actual behavior
- Render the field

### Models and resource files
- The models told above (Dog and Person and their resources)

### System configuration
**Avo version**:
2.35

**Rails version**:
7 and 6 too

**Ruby version**:
3.1.0

**License type**:

 - [X ] Community
 - [X] Pro

**Are you using Avo monkey patches, overriding views or view components?**

 - [ ] Yes. If so, please post code samples.
 - [X] No

### Screenshots or screen recordings

![image](https://github.com/avo-hq/avo/assets/79083991/7f573065-0128-4f48-a417-f7a88e24f2fe)
![image](https://github.com/avo-hq/avo/assets/79083991/de048ae5-e1a1-4cc5-8bb0-92b3fc382119)

#### Impact

 - [ ] High impact (It makes my app un-usable.)
 - [X] Medium impact (I'm annoyed, but I'll live.)
 - [ ] Low impact (It's really a tiny thing that I could live with.)

#### Urgency

 - [ ] High urgency (I can't continue development without it.)
 - [X] Medium urgency (I found a workaround, but I'd love to have it fixed.)
 - [ ] Low urgency (It can wait. I just wanted you to know about it.)

**Comment by adrianthedev** [comment]
Hey @abdellahi-brahim I think I found a bug there. Thanks for the reproduction repo. That really helped out.
I'm working on it and will have a fix next week.

**Comment by Paul-Bob** [comment]
Hi @abdellahi-brahim, what's up!?

Thank you for the detailed description and repo!

As @adrianthedev mentioned, it can be considered a bug. Avo do not allow field id's duplication.

Having:
```ruby
with_options only_on: :index do
  field :id, as: :id
  field :name, as: :text
  field :dogs, as: :tags do |model, resource, view|
    model.dogs.pluck(:name)
  end
end

with_options only_on: :show do
  field :id, as: :id
  field :name, as: :text
  field :dogs, as: :has_many
end
```

it's equal to having:
```ruby
field :id, as: :id, only_on: :index
field :name, as: :text, only_on: :index
field :dogs, as: :tags, only_on: :index do |model, resource, view|
  model.dogs.pluck(:name)
end

field :id, as: :id, only_on: :show
field :name, as: :text, only_on: :show
field :dogs, as: :has_many, only_on: :show
```

This means that the fields `id`, `name,` and `dogs` have multiple definitions, which can lead to conflicts like the one you are currently experiencing. We are actively working on finding a solution for this issue. However, if this problem is hindering the development of your project, I have a suggestion to help you overcome this obstacle.

You can change the id of your `dogs` field but keep same interface using I18n:
```ruby
with_options only_on: :index do
  field :id, as: :id
  field :name, as: :text
  field :dogs_tags, as: :tags, translation_key: "avo.field_translations.dogs" do |model, resource, view|
    model.dogs.pluck(:name)
  end
end

with_options only_on: :show do
  field :id, as: :id
  field :name, as: :text
  field :dogs, as: :has_many
end
```

This should temporary fix and unblock your development.

**Comment by abdellahi-brahim** [comment]
Thank you so much @Paul-Bob ! I will be using that meanwhile

**Comment by adrianthedev** [comment]
Hey @abdellahi-brahim.
I talked it over with @Paul-Bob and figured that this can be fixed using this workaround (changing the order). With Avo 3 around the block, we're reworking a few things there and fixing it in the process.
I'll mark this as a won't fix for the moment so we can focus on bigger & better things.
Thanks!
